crates/runner/src/lib.rs

```diff
@@ -432,63 +432,49 @@ impl Runner {
-                    && f.results().len() == 1
-                    && f.results().all(|r| r.is_i32());
-            }
-
-            if i.module() == instance_name && i.name() == "eval_bytecode" {
-                let ty = i.ty();
-                let f = ty.unwrap_func();
-                return f.params().all(|p| p.is_i32())
-                    && f.params().len() == 2
-                    && f.results().len() == 0;
-            }
-
-            if i.module() == instance_name && i.name() == "memory" {
-                let ty = i.ty();
-                return ty.memory().is_some();
-            }
-
-            false
-        });
-
-        let count = result.count();
-        if count == 3 {
-            Ok(())
-        } else {
-            Err(anyhow!("Unexpected number of imports: {}", count))
-        }
-    }
-
-    pub fn assert_known_named_function_imports(&self) -> Result<()> {
-        self.assert_known_base_imports()?;
-
-        let module = Module::from_binary(self.linker.engine(), &self.wasm)?;
-        let instance_name = self.plugin.namespace();
-        let result = module.imports().filter(|i| {
-            if i.module() == instance_name && i.name() == "invoke" {
-                let ty = i.ty();
-                let f = ty.unwrap_func();
-
-                return f.params().len() == 4
-                    && f.params().all(|p| p.is_i32())
-                    && f.results().len() == 0;
-            }
+        let imports = module
+            .imports()
+            .filter(|i| i.module() == instance_name)
+            .collect::<Vec<_>>();
+        assert_eq!(4, imports.len());
+
+        let realloc = imports
+            .iter()
+            .find(|i| i.name() == "canonical_abi_realloc")
+            .expect("Should have canonical_abi_realloc import");
+        let ty = realloc.ty();
+        let f = ty.unwrap_func();
+        assert!(f.params().all(|p| p.is_i32()));
+        assert_eq!(4, f.params().len());
+        assert!(f.results().all(|p| p.is_i32()));
+        assert_eq!(1, f.results().len());
+
+        let memory = imports
+            .iter()
+            .find(|i| i.name() == "memory" && i.ty().memory().is_some());
+        assert!(memory.is_some(), "Should have memory import");
+
+        let eval_bytecode = imports
+            .iter()
+            .find(|i| i.name() == "eval_bytecode")
+            .expect("Should have eval_bytecode import");
+        let ty = eval_bytecode.ty();
+        let f = ty.unwrap_func();
+        assert!(f.params().all(|p| p.is_i32()));
+        assert_eq!(2, f.params().len());
+        assert_eq!(0, f.results().len());
+
+        let invoke = imports
+            .iter()
+            .find(|i| i.name() == "invoke")
+            .expect("Should have eval_bytecode import");
+        let ty = invoke.ty();
+        let f = ty.unwrap_func();
+        assert!(f.params().all(|p| p.is_i32()));
+        assert_eq!(4, f.params().len());
+        assert_eq!(0, f.results().len());
```

Comment:
Could you replace the `assert`s with `bail!` or `Err(anyhow::Error)`? I get that we're asserting and that the runner is mostly used in tests, however, I'd like to ensure that errors are recoverable here so that failures can bubble up with as much as context as possible to the test invoking them.